pygame 2.0.0 (SDL 2.0.12, python 3.7.10)
Hello from the pygame community. https://www.pygame.org/contribute.html
boxing_v1 pettingzoo
Load boxing_v1 environment in type pettingzoo.
Env observation space: Box(0.0, 1.0, (130,), float32) action space: Discrete(18)
<env.wrappers.mars_wrappers.Dict2TupleWrapper object at 0x7fe163c96350>
boxing_v1 pettingzoo
Load boxing_v1 environment in type pettingzoo.
Env observation space: Box(0.0, 1.0, (130,), float32) action space: Discrete(18)
Agents No. [1] (index starting from 0) are not learnable.
Arguments:  {'env_name': 'boxing_v1', 'env_type': 'pettingzoo', 'num_envs': 1, 'ram': True, 'seed': 1122, 'algorithm': 'DQN', 'algorithm_spec': {'episodic_update': False, 'dueling': False, 'replay_buffer_size': '1e5', 'gamma': 0.99, 'multi_step': 1, 'target_update_interval': 1000, 'eps_start': 1.0, 'eps_final': 0.01, 'eps_decay': 30000}, 'batch_size': 32, 'max_episodes': 60000, 'max_steps_per_episode': 10000, 'train_start_frame': 0, 'optimizer': 'adam', 'learning_rate': '1e-4', 'device': 'gpu', 'update_itr': 1, 'log_avg_window': 20, 'log_interval': 20, 'render': False, 'test': False, 'exploit': False, 'load_model_idx': False, 'load_model_full_path': False, 'save_path': '', 'save_interval': 2000, 'net_architecture': {'hidden_dim_list': [64, 64, 64], 'hidden_activation': 'ReLU', 'output_activation': False}, 'marl_method': 'nxdo2', 'marl_spec': {'selfplay_score_delta': 60, 'trainable_agent_idx': 0, 'opponent_idx': 1}}
Traceback (most recent call last):
  File "general_run.py", line 43, in <module>
    launch_rollout(parser_args.env, parser_args.method)
  File "general_run.py", line 37, in launch_rollout
    rollout(env, model, args)
  File "/home/zihan/research/MARS/mars_core/rollout.py", line 19, in rollout
    rollout_normal(env, model, args)
  File "/home/zihan/research/MARS/mars_core/rollout.py", line 124, in rollout_normal
    model, logger, env, args
  File "/home/zihan/research/MARS/mars_core/marl/double_oracle.py", line 194, in step
    score_delta = np.mean(logger.epi_rewards[logger.keys[self.current_learnable_model_idx]][-score_avg_window:])\
TypeError: list indices must be integers or slices, not str
